src/prefect/engine/task_runner.py

```diff
@@ -577,7 +577,7 @@ def check_task_is_cached(self, state: State, inputs: Dict[str, Result]) -> State
                 state._result = state._result.to_result()
                 return state
             else:
-                self.logger.debug(
+                self.logger.warning(
```

Comment:
Open to push back on this change; it will always happen _eventually_, but knowing _when_ it happens seems important so I elevated this log level to "WARNING".